src/index.ts

```diff
@@ -75,9 +75,12 @@ export default function useSound(
     }
     // The linter wants to run this effect whenever ANYTHING changes,
     // but very specifically I only want to recreate the Howl instance
-    // when the `url` changes. Other changes should have no effect.
+    // when the `src` changes. Other changes should have no effect.
+    // Passing array to the useEffect dependencies list will result in
+    // ifinite loop so we need to stringify it, for more details check
+    // https://github.com/facebook/react/issues/14476#issuecomment-471199055
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [url]);
+  }, [JSON.stringify(src)]);
```

Comment:
Another alternative here (so long as you ensure `src` is an array first) would be to spread the `src` array into the deps (or since it appears to be the only dep, you could just use it directly:

```suggestion
  }, [...src]);
```

```suggestion
  }, src);
```

Reply:
I think the stringify solution works well in this case!